.taskcluster.yml

```diff
@@ -1,102 +1,100 @@
-            npm run lint &&
-            npm run package &&
-            /repo/bin/commit_xpi.sh
-      metadata:
-        name: "Firefox Experiments GitHub Tests"
-        description: "All tests"
-        owner: ${event.pull_request.user.login}@users.noreply.github.com
-        source: ${event.repository.url}
-  # When a commit is pushed to master, perform this task
-  # Note: Currently this is wrongly showing a status on PR pushes in GitHub (Bug 1514914)
-  # If a branch is merged without being updated first, the tests could pass on the branch
-  # but not when things are merged into master.
-  - $if: 'tasks_for == "github-push"'
-    then:
-      $if: 'event.ref == "refs/heads/master"'
+  $let:
+    # This only supports pushes and pull requests for now. If we start responding to another
+    # action type, make this a $match.
+    repo:
+      $if: 'tasks_for == "github-push"'
       then:
-        taskId: {$eval: as_slugid("push_task")}
+        git_url: ${event.repository.url}
+        url: ${event.repository.url}
+        ref: ${event.after}
+        owner: ${event.pusher.name}
+      else:
+        git_url: ${event.pull_request.head.repo.git_url}
+        url: ${event.pull_request.head.repo.url}
+        ref: ${event.pull_request.head.sha}
+        owner: ${event.pull_request.user.login}
+  in:
+    $let:
+      taskRef:
+        # TC creates a task group with tasks in it; each task has an ID. TC will provide a link
+        # built with these IDs in GH's GUI every time a task is created, even if it fails.
+        taskId: {$eval: as_slugid("pr_task")} # Returns a hash of a string
         created: {$fromNow: ''}
+        # If the task is not completed by its deadline, it will be resolved as `exception`
+        # with reason "deadline-exceeded"
         deadline: {$fromNow: '1 hour'}
+        # Identifies the Taskcluster provisioner responsible for the compute resources
+        # (could be 1 CPU, a fraction of a CPU, multiple CPUs...)that will execute the
+        # task. e.g. 'aws-provisioner-v1' is the AWS Provisioner, which runs its tasks
+        # on Amazon EC2 instances using Docker.
         provisionerId: aws-provisioner-v1
-        workerType: github-worker
-        payload:
-          maxRunTime: 600
-          image: "node:8"
-          artifacts:
+        # A parameter specific to the AWS Provisioner which identifies the pool of EC2
+        # resources within which the task should be executed. Pools may use different
+        # EC2 instance types, AMIs (Amazon Machine Images), etc.
+        # Generally, github-worker is a variant of docker-worker in TC docs
+        workerType: github-worker # TC's generic worker for any github projects
+        # A scope is a permission to perform a particular action. A task has a set of scopes
+        # that determine what the task can do. These scopes are validated by the worker.
+        scopes:
+          - secrets:get:github.com/biancadanforth/taskcluster-integration-poc/ssh-key
```

Comment:
I imagine this repository will be moved under one of the Mozilla organization soon, at which point those scopes will no longer reference your personal github. Is that correct?

Reply:
Yes, that is covered by Issue #15 , and issue #13 .